contrib/boot-contracts-stateful-prop-tests/tests/pox-4/err_Commands.ts

```diff
@@ -0,0 +1,2283 @@
+      extendCount: fc.integer({ min: 1, max: 11 }),
+    }).chain((r) => {
+      const operator = stackers.get(r.operator.stxAddress)!;
+      const delegatorsList = operator.poolMembers;
+      const availableStackers = delegatorsList.filter((delegator) => {
+        const delegatorWallet = stackers.get(delegator)!;
+        return delegatorWallet.unlockHeight > nextCycleFirstBlock(network);
+      });
+
+      const availableStackersOrFallback = availableStackers.length === 0
+        ? [r.operator.stxAddress]
+        : availableStackers;
+
+      return fc
+        .record({
+          stacker: fc.constantFrom(...availableStackersOrFallback),
+          currentCycle: fc.constant(currentCycle(network)),
+        })
+        .map((additionalProps) => ({
+          ...r,
+          stacker: wallets.get(additionalProps.stacker)!,
+          currentCycle: additionalProps.currentCycle,
+        }));
+    }).map(
+      (final) =>
+        new DelegateStackExtendCommand_Err(
+          final.operator,
+          final.stacker,
+          final.extendCount,
+          final.currentCycle,
+          function (
+            this: DelegateStackExtendCommand_Err,
+            model: Readonly<Stub>,
+          ): boolean {
+            const operatorWallet = model.stackers.get(
+              this.operator.stxAddress,
+            )!;
+            const stackerWallet = model.stackers.get(
+              this.stacker.stxAddress,
+            )!;
+
+            const firstRewardCycle =
+              this.currentCycle > stackerWallet.firstLockedRewardCycle
+                ? this.currentCycle
+                : stackerWallet.firstLockedRewardCycle;
+            const firstExtendCycle = Math.floor(
+              (stackerWallet.unlockHeight - FIRST_BURNCHAIN_BLOCK_HEIGHT) /
+                REWARD_CYCLE_LENGTH,
+            );
+            const lastExtendCycle = firstExtendCycle + this.extendCount - 1;
+            const totalPeriod = lastExtendCycle - firstRewardCycle + 1;
+            const newUnlockHeight =
+              REWARD_CYCLE_LENGTH * (firstRewardCycle + totalPeriod - 1) +
+              FIRST_BURNCHAIN_BLOCK_HEIGHT;
+            const stackedAmount = stackerWallet.amountLocked;
+
+            if (
+              stackerWallet.amountLocked > 0 &&
+              !(stackerWallet.hasDelegated === true) &&
+              stackerWallet.isStacking === true &&
+              stackerWallet.isStackingSolo === true &&
+              !(stackerWallet.delegatedTo === this.operator.stxAddress) &&
+              !(stackerWallet.delegatedUntilBurnHt === undefined ||
+                stackerWallet.delegatedUntilBurnHt >= newUnlockHeight) &&
+              !(stackerWallet.delegatedMaxAmount >= stackedAmount) &&
+              !operatorWallet.poolMembers.includes(this.stacker.stxAddress) &&
+              !operatorWallet.lockedAddresses.includes(
+                this.stacker.stxAddress,
+              ) &&
+              totalPeriod <= 12
```

Comment:
I think that huge conditions like that should be split into smaller variables `isStacking` `isDelegating` etc

Or it will be nightmare to debug/refactor in the future

Reply:
Done ✅